WalletWasabi.Fluent/Models/WebcamQrReader.cs

```diff
@@ -0,0 +1,160 @@
+using Avalonia.Media.Imaging;
+using System;
+using System.Threading.Tasks;
+using OpenCvSharp;
+using Avalonia;
+using Avalonia.Media;
+using System.Runtime.InteropServices;
+using WalletWasabi.Logging;
+using WalletWasabi.Userfacing;
+using NBitcoin;
+using Nito.AsyncEx;
+using Avalonia.Platform;
+
+namespace WalletWasabi.Fluent.Models
+{
+	public class WebcamQrReader
+	{
+		private const byte DefaultCameraId = 0;
+
+		public WebcamQrReader(Network network)
+		{
+			ScanningTaskLock = new();
+			Network = network;
+		}
+
+		public event EventHandler<WriteableBitmap>? NewImageArrived;
+
+		public event EventHandler<string>? CorrectAddressFound;
+
+		public event EventHandler<string>? InvalidAddressFound;
+
+		public event EventHandler<Exception>? ErrorOccured;
+
+		private AsyncLock ScanningTaskLock { get; set; }
+		public bool RequestEnd { get; set; }
+		public Network Network { get; }
+		public Task? ScanningTask { get; set; }
+		public bool IsRunning => ScanningTask is not null;
+
+		public async Task StartScanningAsync()
+		{
+			using (await ScanningTaskLock.LockAsync().ConfigureAwait(false))
+			{
+				if (!RuntimeInformation.IsOSPlatform(OSPlatform.Windows))
+				{
+					ErrorOccured?.Invoke(this, new NotImplementedException("This operating system is not supported."));
+				}
+				ScanningTask = Task.Run(() =>
+				{
+					VideoCapture? camera = null;
+					try
+					{
+						camera = new();
```

Comment:
Consider using `camera.SetExceptionMode(true);` It will throw exception on error instead of showing error codes.
i.e.: You can detect by this if the camera is in use, although I didn't test it too much.